tensorflow_datasets/image/tiered_imagenet.py

```diff
@@ -0,0 +1,137 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl import logging
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+import io
+import os
+import csv
+import numpy as np
+import gzip
+import pickle as pkl
+
+class TieredImagenet(tfds.core.GeneratorBasedBuilder):
+    """tiered_imagenet is larger subset  of Imagenet that uses splits defined for benchmarking few-shot learning approaches"""
+    VERSION = tfds.core.Version("1.0.2")
+
+    _CITATION = """\
+        @inproceedings{ren18fewshotssl,
+        author   = {Mengye Ren and 
+                    Eleni Triantafillou and 
+                    Sachin Ravi and 
+                    Jake Snell and 
+                    Kevin Swersky and 
+                    Joshua B. Tenenbaum and 
+                    Hugo Larochelle and 
+                    Richard S. Zemel},
+        title    = {Meta-Learning for Semi-Supervised Few-Shot Classification},
+        booktitle= {Proceedings of 6th International Conference on Learning Representations {ICLR}},
+        year     = {2018},
+        }
+    """
+    _DESCRIPTION = """\

```

Comment:
CITATION, DESCRIPTION etc. should be defined top of the class to  be consistent with other datasets.

Reply:
Thanks.  I will change it. Can you please review the fake example generation as well?